stdlib/public/Concurrency/Task.cpp

```diff
@@ -534,32 +561,93 @@ static AsyncTaskAndContext swift_task_create_commonImpl(
   }
 
   AsyncTask *parent = nullptr;
+  AsyncTask *currentTask = swift_task_getCurrent();
   if (jobFlags.task_isChildTask()) {
-    parent = swift_task_getCurrent();
+    parent = currentTask;
     assert(parent != nullptr && "creating a child task with no active task");
   }
 
-  // Inherit the priority of the currently-executing task if unspecified and
-  // we want to inherit.
-  if (jobFlags.getPriority() == JobPriority::Unspecified &&
-      (jobFlags.task_isChildTask() || taskCreateFlags.inheritContext())) {
-    AsyncTask *currentTask = parent;
-    if (!currentTask)
-      currentTask = swift_task_getCurrent();
-
-    if (currentTask)
-      jobFlags.setPriority(currentTask->getPriority());
-    else if (taskCreateFlags.inheritContext())
-      jobFlags.setPriority(swift_task_getCurrentThreadPriority());
+  // Start with user specified priority at creation time (if any)
+  JobPriority basePriority = (taskCreateFlags.getCreationPriority());
+
+
+  if (swift_task_is_detached(taskCreateFlags, jobFlags)) {
+    // Case 1: No priority specified
+    //    Base priority = UN
+    //    Escalated priority = UN
+    // Case 2: Priority specified
+    //    Base priority = user specified priority
+    //    Escalated priority = UN
+    //
+    // Task will be created with max priority = max(base priority, UN) = base
```

Comment:
I don't really understand the point of all this math with `unspecified`.  Neither the creation nor the escalated priority should actually be `unspecified` on a live task.  I feel like, within these blocks, we should be deriving the real base priority, such that it's a postcondition of this if/else that `basePriority != unspecified`.

Reply:
> Neither the creation nor the escalated priority should actually be unspecified on a live task.

Agreed and isn't that what we're doing? We're starting the basePriority with what was specified at task creation time (as in line 571), which will be Unspecified if the user doesn't specify anything. This logic here does the additional propagation logic such that by the end of it, we end up with basePriority != UN.

The reason I am drawing the distinction here, is cause we only want to apply the UI -> IN downgrade during automatic propagation and not if the user explicitly asked for UI priority.